doc/ug/getting_started_hw_design.md

```diff
@@ -90,3 +90,42 @@ To do so, follow the steps below.
 * Open a Pull Request with the necessary code changes.
 
 If it is not clear on how to proceed, feel free to file an issue requesting assistance.
+
+## Change of Interrupts
+
+Change of the interrupts in a HWIP needs lots of extra works.
+DV environment, if exists, should have the newly introduced, or renamed ports in its TB module.
+TOP module, if the module has been instantiated, should be re-generated to have new or renamed ports.
+DIF SW should be regenerated too.
+The PLIC test should be updated also.
+
+### Update TOP
+
+```console
+$ make -C hw top
+```
+
+After you revise the IP `.hjson`, IP top module, IP register interface, and DV testbench `tb.sv`, you can re-generate top earlgrey module with above command.
+It reads the `.hjson` file and update the interrupt signal and re-generate PLIC module if needed.
+
+### New DIF
+
+Unfortunately, re-generating TOP does not resolve everything.
+If the interrupt name is changed or new interrupt is introduced, the DIF for the IP should be re-generated.
+
+```console
+$ ./util/make_new_dif.py --mode=regen --only=autogen
+```
+
+The command above updates the DIF (auto-generated part) under `sw/device/lib/dif/`.
+
+If DIF for the HWIP already uses the interrupt enums inside, you need to manually revise the reference.
+In most cases, the interrupt has *kDif* then HWIP name in PascalCase (e.g *SpiDevice*), *Irq*, then the interrupt name in PascalCase (e.g. *GenericRxFull*).
+Also, if DIF refers Top Plic interrupt enum, the format is *kTopEarlgreyPlicIrq*, then HWIP name in PascalCase, then the interrupt name in PascalCase.
+
+### PLIC Unittest
+
+If the number of interrupts is changed, you need to revise the SW unittest manually.
+
+Open `sw/device/lib/dif/dif_rv_plic_unittest.cc` and change the `RV_PLIC_PARAM_NUM_SRC` macro to the current interrupt number.
```

Comment:
(Not an actionable item, just commenting to document this / through this thought out there): would be nice (in the future) if we just parameterized this whole DIF/unittest combo and autogened the whole thing from the HJSONs.

Reply:
Hear, hear!